pypeit/edgetrace.py

```diff
@@ -0,0 +1,4109 @@
+
+        # TODO: It's possible for the master key and the detector
+        # number to be inconsistent...
+        masterframe.MasterFrame.__init__(self, self.master_type, master_dir=master_dir,
+                                         master_key=master_key)
+
+        self.spectrograph = spectrograph    # Spectrograph used to take the data
+        self.par = par                      # Parameters used for slit edge tracing
+
+        self.files = None               # Files used to construct the trace image
+        self.img = None                 # The image used to find the slit edges
+        self.bpm = None                 # Mask for the trace image
+        self.det = None                 # Detector used for the trace image
+        self.binning = None             # Detector ordered spectral then spatial
+        self.sobel_sig = None           # Sobel filtered image used to detect edges
+        self.sobel_sig_left = None      # Sobel filtered image used to trace left edges
+        self.sobel_sig_right = None     # Sobel filtered image used to trace right edges
+        # TODO: Need a separate mask for the sobel image?
+        self.nspec = None               # The shape of the trace image is (nspec,nspat)
+        self.nspat = None
+
+        self.traceid = None             # The ID numbers for each trace
+        self.spat_img = None            # (Integer) Pixel nearest the slit edge for each trace
+        self.spat_cen = None            # (Floating-point) Spatial coordinate of the slit edges
+                                        # for each spectral pixel
+        self.spat_err = None            # Error in the slit edge spatial coordinate
+        self.spat_msk = None            # Mask for the slit edge position for each spectral pixel
+
+        self.spat_fit = None            # The result of modeling the slit edge positions
+        self.spat_fit_type = None       # The type of fitting performed
+        
+        self.pca = None                 # One or two EdgeTracePCA objects with PCA decomposition
+        self.pca_type = None            # Measurements used to construct the PCA (center or fit)
+
+        self.design = None              # Table that collates slit-mask design data matched to
+                                        # the edge traces
+        self.objects = None             # Table that collates object information, if available
+                                        # in the slit-mask design, matched to the `design` table.
+
+        self.qa_path = qa_path          # Directory for QA output
+
+        self.log = None                 # Log of methods applied
+
+        if img is not None and load:
+            msgs.error('Arguments img and load are mutually exclusive.  Choose to either trace '
+                       'a new image or load a previous trace.')
+
+        if load:
+            # Attempt to load an existing master frame
+            self.load()
```

Comment:
It faults.  Hmm... appreciating now that this is different behavior than all the other `MasterFrames`...  The way I handle this in `calibrations.py` is:
```
if self.reuse_masters and self.edges.exists():
            self.edges.load()
```
So it won't try to load if the output file doesn't exist.  The reason I didn't do the normal thing of returning `None` if the load is not possible is because load would naturally not return anything if it was successful.  Personally, I'd rather have methods fault if they're called and fail, but I see the conflict with the rest of the code.  Thoughts?